ldms/src/sampler/Makefile.am

```diff
@@ -13,6 +13,10 @@ CORE_LIBADD = $(top_builddir)/ldms/src/core/libldms.la \
 COMMON_LIBADD = libsampler_base.la $(CORE_LIBADD) \
 		$(top_builddir)/lib/src/coll/libcoll.la
 
+libldms_compid_helper_la_SOURCES = comp_id_helper.c comp_id_helper.h
+libldms_compid_helper_la_LIBADD = $(CORE_LIBADD)
+lib_LTLIBRARIES += libldms_compid_helper.la
```

Comment:
I am leery of adding these as exported interfaces until we do a little more design work. jobid_helper was intended as a stop gap implementation until we figure out a consistent API to offer external developers. I think this should be noinst_LTLIBARIES like libjobid_helper, no lib_LTLIBRARIES.

Reply:
Whether provided as inst or noist, it's still an API. Why take the binary bloat? We're about to add a whole new API anyway, so I'm not seeing the harm.

Reply:
In the past, we had another solution to the jobid helper problem.  The problem is either you look up the job data set *every time*, or you have an awful assumption that the jobid set pointer is not going to expire underneath you. The alternate solution was the (inevitable?) global registry of interesting data items. A plugin/instance can in theory come and go, independent of the registry. The registry can contain named values, such as the stuff we're doing with jobid.   The counter-argument can be made that the local list of set instance is is just another registry, if a slower one because you must lookup the set and then find its element index and then read that value.

Reply:
> Whether provided as inst or noist, it's still an API. Why take the binary bloat? We're about to add a whole new API anyway, so I'm not seeing the harm.

The increase to binary sizes is insignificant. Adding exporting a public interface imposes maintenance burdens on us as a code team (and if we ignore those burdens, then we produce less-than-pleasant to work with software). Internal-only interfaces can be changed as we like without breaking anything. External facing APIs and ABIs cannot be changed without carefully publishing and orchestrating change.

I do not think we should be exporting these tiny helper libraries. A better approach would be to wrap them up into a larger consistent API and single library. So until we want to go the work of designing a good externally facing API, lets just keep it internal.

Reply:
> In the past, we had another solution to the jobid helper problem.

I honestly can't tell what this has to do with this PR.

Reply:
@morrone you mentioned jobid_helper as a stopgap.  It would have been nice (and maybe in the new api still could be) if we put component_id and all the job stuff into a registry instead of dragging it all through config lines.

Reply:
Still need to change to noinst.